Snowplow/Internal/Tracker/SPStateFuture.m

```diff
@@ -0,0 +1,59 @@
+//
+//  SPStateUpdater.m
+//  Snowplow
+//
+//  Copyright (c) 2013-2021 Snowplow Analytics Ltd. All rights reserved.
+//
+//  This program is licensed to you under the Apache License Version 2.0,
+//  and you may not use this file except in compliance with the Apache License
+//  Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+//  http://www.apache.org/licenses/LICENSE-2.0.
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the Apache License Version 2.0 is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+//  express or implied. See the Apache License Version 2.0 for the specific
+//  language governing permissions and limitations there under.
+//
+//  Authors: Alex Benini
+//  Copyright: Copyright (c) 2013-2021 Snowplow Analytics Ltd
+//  License: Apache License Version 2.0
+//
+
+#import "SPStateFuture.h"
+
+
+@interface SPStateFuture ()
+
+@property (nonatomic) SPEvent *event;
+@property (nonatomic) SPStateFuture *previousState;
```

Comment:
True. The Future suffix should indicate that the class works as indicated in theory around Futures and Promises (https://en.wikipedia.org/wiki/Futures_and_promises).
I called it StateFuture because theoretically it should have a computed state until we explicitly ask the state value.
So it doesn't represent a future state but a state value ready in the future.
Happy to better suggestions.

Reply:
Yeah, I figured you were using a `Futures` concept here, I just find it reads strange. I think its fine but maybe a comment or two in SPStateFuture is worth it to make it clearer.